### PR TITLE
Update repository to better match our most recently changed packages

### PR DESCRIPTION
**Pull request by ciaransweet**
This PR does a fair bit of quality of life improvements to the `chaosspring` package.

Notably it:
* Switches from TravisCI to GitHub Actions
* Uses the same linting & contribution styles as more up-to-date packages (`flake8` et al, adding Makefiles etc)
* Updates the README
* Removes Py35 support

### CODE REVIEW

**Review by Lawouach — DISMISSED** [review]
LGTM

File: chaosspring/actions.py [hunk 1]
```diff
@@ -7,81 +6,91 @@
...
+        assaults_configuration=assaults_configuration,
+        headers=headers,
+        timeout=timeout,
+        configuration=configuration,
+        secrets=secrets,
+    )
 
     if response.status_code != codes.ok:
         raise FailedActivity(
             "Change ChaosMonkey Assaults Configuration failed: {m}".format(
```
Comment (CharlieMoon37) [inline chaosspring/actions.py]: For some reason, it doesn't look like pyupgrade has worked here as we are still using `format`?

File: chaosspring/api.py [hunk 2]
```diff
@@ -34,7 +36,8 @@ def call_api(base_url: str,
     """
 
     url = "{base_url}/{api_endpoint}".format(
```
Comment (CharlieMoon37) [inline chaosspring/api.py]: Ditto

**Review by CharlieMoon37 — APPROVED** [review]
Lovely Stuff!

**Pull request merged**